src/test/unit/StrategyManagerUnit.t.sol

```diff
@@ -1075,27 +830,260 @@ contract StrategyManagerUnitTests is Test, Utils {
+        uint256 depositAmount,
+        uint256 removeSharesAmount
+    ) external {
+        cheats.assume(staker != address(0));
+        cheats.assume(depositAmount > 0 && depositAmount < dummyToken.totalSupply());
+        cheats.assume(removeSharesAmount > depositAmount);
+        IStrategy strategy = dummyStrat;
+        _depositIntoStrategySuccessfully(strategy, staker, depositAmount);
+        cheats.expectRevert(bytes("StrategyManager._removeShares: shareAmount too high"));
+        delegationManagerMock.removeShares(strategyManager, staker, strategy, removeSharesAmount);
+    }
+
+    function testRemoveSharesRemovesStakerStrategyListSingleStrat(address staker, uint256 sharesAmount) external {
+        cheats.assume(staker != address(0));
+        cheats.assume(sharesAmount > 0 && sharesAmount < dummyToken.totalSupply());
+        IStrategy strategy = dummyStrat;
+        _depositIntoStrategySuccessfully(strategy, staker, sharesAmount);
+
+        uint256 stakerStrategyListLengthBefore = strategyManager.stakerStrategyListLength(staker);
+        uint256 sharesBefore = strategyManager.stakerStrategyShares(staker, strategy);
+        require(sharesBefore == sharesAmount, "Staker has not deposited amount into strategy");
+
+        delegationManagerMock.removeShares(strategyManager, staker, strategy, sharesAmount);
+        uint256 stakerStrategyListLengthAfter = strategyManager.stakerStrategyListLength(staker);
+        uint256 sharesAfter = strategyManager.stakerStrategyShares(staker, strategy);
+        require(
+            stakerStrategyListLengthAfter == stakerStrategyListLengthBefore - 1,
+            "stakerStrategyListLengthAfter != stakerStrategyListLengthBefore - 1"
+        );
+        require(sharesAfter == 0, "sharesAfter != 0");
+        require(!_isDepositedStrategy(staker, strategy), "strategy should not be part of staker strategy list");
+    }
+
+    // Remove Strategy from staker strategy list with multiple strategies deposited
+    function testRemoveSharesRemovesStakerStrategyListMultipleStrat(
+        address staker,
+        uint256[3] memory amounts,
+        uint8 randStrategy
+    ) external {
+        cheats.assume(staker != address(0));
+        cheats.assume(amounts[0] > 0 && amounts[0] < dummyToken.totalSupply());
+        cheats.assume(amounts[1] > 0 && amounts[1] < dummyToken.totalSupply());
+        cheats.assume(amounts[2] > 0 && amounts[2] < dummyToken.totalSupply());
+        IStrategy[] memory strategies = new IStrategy[](3);
+        strategies[0] = dummyStrat;
+        strategies[1] = dummyStrat2;
+        strategies[2] = dummyStrat3;
+        IStrategy removeStrategy = strategies[randStrategy % 3];
+        uint256 removeAmount = amounts[randStrategy % 3];
+        _depositIntoStrategySuccessfully(strategies[0], staker, amounts[0]);
+        _depositIntoStrategySuccessfully(strategies[1], staker, amounts[1]);
+        _depositIntoStrategySuccessfully(strategies[2], staker, amounts[2]);
+
```

Comment:
could put some of this in a loop, but it's fine as-is honestly

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/249/commits/c2eb3ff71905ebd4c3293471201e204e6f54f7f1 Looops